controllers/depresolver/depresolver_test.go

```diff
@@ -92,21 +92,22 @@ func TestResolveSpecWithEmptyFields(t *testing.T) {
 func TestResolveSpecWithNegativeFields(t *testing.T) {
 	// arrange
 	cl, gslb := getTestContext("./testdata/invalid_omitempty_negative.yaml")
-	resolver := NewDependencyResolver(context.TODO(), cl)
+	resolver := NewDependencyResolver(cl)
 	// act
-	err := resolver.ResolveGslbSpec(gslb)
+	err := resolver.ResolveGslbSpec(context.TODO(), gslb)
```

Comment:
Why do we still use TODO() there

Reply:
@ytsarev good question: The point is that you are creating i.e. WithCancel context from existing context.

 If you use Background context, you explicitly say: Hey, I'm using context from the top of the tree, usually `main` and this context will instantiate another context(s) afterwards `WithCancel`, `WithDeadline`... 

If you use context.TODO() you explicitly say: function just requires some context argument so I will pass TODO which is doing nothing, just to compile. For instance, our tests..

Reply:
@kuritka cool, makes total sense